packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -2464,6 +2489,397 @@ class _MenuDirectionalFocusAction extends DirectionalFocusAction {
+/// not necessary to provide one.
+///
+/// {@template flutter.material.MenuAcceleratorLabel.accelerator_sample}
+/// {@tool dartpad} This example shows a [MenuBar] that handles keyboard
+/// accelerators using [MenuAcceleratorLabel]. To use the accelerators, press
+/// the Alt key to see which letters are underlined in the menu bar, and then
+/// press the appropriate letter. Accelerators are not supported on macOS or iOS
+/// since those platforms don't support them natively, so this demo will only
+/// show a regular Material menu bar on those platforms.
+///
+/// ** See code in
+/// examples/api/lib/material/menu_anchor/menu_accelerator_label.0.dart **
+/// {@end-tool}
+/// {@endtemplate}
+class MenuAcceleratorLabel extends StatefulWidget {
+  /// Creates a const [MenuAcceleratorLabel].
+  ///
+  /// The [label] parameter is required.
+  const MenuAcceleratorLabel(
+    this.label, {
+    super.key,
+    this.builder = defaultLabelBuilder,
+  });
+
+  /// The label string that should be displayed.
+  ///
+  /// The label string provides the label text, as well as the possible
+  /// characters which could be used as accelerators in the menu system.
+  ///
+  /// {@template flutter.material.menu_anchor.menu_accelerator_label.label}
+  /// To indicate which letters in the label are to be used as accelerators, add
+  /// a "&" character before the character in the string. If more than one
+  /// character has an "&" in front of it, then the characters appearing earlier
+  /// in the string are preferred. To represent a literal "&", insert "&&" into
+  /// the string. All other ampersands will be removed from the string before
+  /// calling [MenuAcceleratorLabel.builder].
+  /// {@endtemplate}
+  ///
+  /// See also:
+  ///
+  /// * [displayLabel], which returns the [label] with all of the ampersands
+  ///   stripped out of it, and double ampersands converted to ampersands.
+  /// * [stripAcceleratorMarkers], which returns the supplied string with all of
+  ///   the ampersands stripped out of it, and double ampersands converted to
+  ///   ampersands, and optionally returns the index of the accelerator
+  ///   character found.
+  final String label;
+
+  /// Returns the [label] with any accelerator markers removed.
+  ///
+  /// This getter just calls [stripAcceleratorMarkers] with the [label].
+  String get displayLabel => stripAcceleratorMarkers(label);
+
+  /// The optional [MenuAcceleratorChildBuilder] which is used to build the
+  /// widget that displays the label itself.
+  ///
+  /// The [defaultLabelBuilder] function serves as the default value for
+  /// [builder], rendering the label as a [RichText] widget with appropriate
+  /// [TextSpan]s for rendering the label with an underscore under the selected
+  /// accelerator for the label when accelerators have been activated.
+  ///
+  /// {@macro flutter.material.menu_anchor.menu_accelerator_child_builder.args}
+  final MenuAcceleratorChildBuilder builder;
```

Comment:
Nit: If you think it's useful, maybe state here that users don't need to consider the current platform when writing their `builder` function, i.e. they don't need to strip out the `&` on MacOS.

Reply:
Good idea.